File: backjoon/greedy/Python/동전O.py
def solve(N, K):
    answer = 0
    
    coins = []

    for _ in range(N) :
        coins.append(int(input()))#배열에 추가하는 메소드
    
    #reverse True 함으로써 내림차순 정렬
    coins.sort(reverse=True)

    for coin in coins :
        if K >= coin : #몫 만큼 더해주기
            answer += K // coin 
            K %= coin # 나머지 할당
            if K <= 0 :
                break

    print(answer)

if __name__ == "__main__":
    N, K = list(map(int, input().split()))
    solve(N, K)  

File: backjoon/greedy/JS/동전O.js
const input = require("fs")
  .readFileSync(process.platform === "linux" ? "./dev/stdin" : "./input.txt")
  .toString()
  .trim()
  .split("\n");

let [n, k] = input.shift().split(" ").map(Number);
let coins = input.map(Number).sort((a, b) => b - a);
let answer = 0; //동전갯수 구하기 위해 설정

for (let i = 0; i < coins.length; i++) {
  if (k < coins[i]) continue;
  else {
    answer += Math.floor(k / coins[i]);
    k %= coins[i];
  }
}

console.log(answer);
